src/Toml/Codec/Combinator/Custom.hs

```diff
@@ -46,3 +52,89 @@ nullary sum types.
 enumBounded :: (Bounded a, Enum a, Show a) => Key -> TomlCodec a
 enumBounded = match _EnumBounded
 {-# INLINE enumBounded #-}
+
+{- | Codec that checks the 'BiMap' on the given predicate.
+The predicate function returns the value, if the validation is successful, or
+the 'Text' of the error that should be returned in case of validation failure.
+
+__Example:__
```

Comment:
Do you think it's worth it to make this section expandable? I'm afraid, nobody will see it if we make it expandable, and the doc size for this module is still not that big, but maybe you have a different idea

Reply:
I also had the same idea but was convinced that nobody will ever notice the example then 😄 Let's consider this option when the module becomes too long.